File: Swiman/__scripts__/modules/api/models/OrderAuto.ts


    
        import { Season } from "./Season";
    
        import { Category } from "./Category";
    
        import { Activity } from "./Activity";
    
        import { Section } from "./Section";
    

    import * as assign from "object-assign";

    

    


    class OrderAuto {
        

        
        // ID
        public id: number = 0;
        // DATE_FROM
        public date_from: Date = undefined;
        // DATE_TO
        public date_to: Date = undefined;
        // DESCRIPTION
        public description: string = undefined;
        // RESTRICTION_AGE
        public restriction_age: number = undefined;
        // RESTRICTION_AGE_OPERATOR
        public restriction_age_operator: string = undefined;
        // RESTRICTION_DAYS
        public restriction_days: string = undefined;
        // RESTRICTION_MEMBER_NB
        public restriction_member_nb: number = undefined;
        // RESTRICTION_MEMBER_NB_OPERATOR
        public restriction_member_nb_operator: string = undefined;
        // RESTRICTION_ADHESION_NB
        public restriction_adhesion_nb: number = undefined;
        // RESTRICTION_ADHESION_NB_OPERATOR
        public restriction_adhesion_nb_operator: string = undefined;
        // ACTION_AMOUNT_FIX
        public action_amount_fix: number = undefined;
        // ACTION_AMOUNT
        public action_amount: number = undefined;
        // ACTION_AMOUNT_POURCENT
        public action_amount_pourcent: number = undefined;
        // APPLY_ON_MEMBER
        public apply_on_member: boolean = false;
        // APPLY_ON_FAMILY
        public apply_on_family: boolean = false;
        // APPLY_ON_ADHESION
        public apply_on_adhesion: boolean = false;
        // ORDER
        public order: number = 0;
        // SEASON
        public season: Season = undefined;
        // RESTRICTION_CATEGORY
        public restriction_category: Category = undefined;
        // RESTRICTION_ACTIVITY
        public restriction_activity: Activity = undefined;
        // RESTRICTION_SECTION
        public restriction_section: Section = undefined;

        constructor(init?: Partial<OrderAuto>){
            assign(this, init);
        }
        
    }


export {  OrderAuto }

File: Layer_datas/Entities/orders/OrderAuto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Layer_datas.Entities.orders
{
    [Table("orders_auto")]
    public class OrderAuto
    {
        [Key]
        public int id { get; set; }
        public int season_id { get; set; }
        public DateTime date_from { get; set; }
        public DateTime date_to { get; set; }
        public string description { get; set; }

        public int? restriction_category_id { get; set; }
        public int? restriction_activity_id { get; set; }
        public int? restriction_section_id { get; set; }
        public int? restriction_age { get; set; }
        public string restriction_age_operator { get; set; }
        public string restriction_days { get; set; }
        public int? restriction_member_nb { get; set; }
        public string restriction_member_nb_operator { get; set; }
        public int? restriction_adhesion_nb { get; set; }
        public string restriction_adhesion_nb_operator { get; set; }

        public double? action_amount_fix { get; set; }
        public double? action_amount { get; set; }
        public double? action_amount_pourcent { get; set; }

        public bool apply_on_member { get; set; }
        public bool apply_on_family { get; set; }
        public bool apply_on_adhesion { get; set; }

        public int order { get; set; }

        [ForeignKey("season_id")]
        public virtual Season season { get; set; }
        [ForeignKey("restriction_category_id")]
        public virtual activities.Category restriction_category { get; set; }
        [ForeignKey("restriction_activity_id")]
        public virtual activities.Activity restriction_activity { get; set; }
        [ForeignKey("restriction_section_id")]
        public virtual activities.Section restriction_section { get; set; }
    }
}
